server.c++

```diff
@@ -419,27 +421,25 @@ void writeStream(kj::ArrayPtr<RingBuffer> ringBuffers, kj::OutputStream& out,
   uint64_t totalSamples = 0;
   uint samplesSinceFlush = 0;
   for (uint counter = 0; !state.disconnected; counter++) {
-    if (codecId == AV_CODEC_ID_AAC && counter >= 100) {
-      // TODO: I'm only returning 1s of audio for AAC for debugging reasons... once it actually
-      // works, remove the above.
-      KJ_AVCALL(avcodec_send_frame(codecCtx, nullptr));
-    } else {
-      auto samples = kj::arrayPtr(reinterpret_cast<float*>(frame->data[0]), frame->nb_samples);
-      for (auto i: kj::zeroTo(ringBuffers.size())) {
-        // Don't get more than 1s behind.
-        size_t maxLag = inputOffsets[i] == 0 ? 0 : frame->sample_rate * sizeof(float);
-        inputOffsets[i] = ringBuffers[i].read(inputOffsets[i], maxLag, samples.asBytes(), i > 0);
-      }
+    auto samples = kj::arrayPtr(reinterpret_cast<float*>(frame->data[0]), frame->nb_samples);
+    for (auto i: kj::zeroTo(ringBuffers.size())) {
+      // Don't get more than 1s behind.
+      size_t maxLag = inputOffsets[i] == 0 ? 0 : frame->sample_rate * sizeof(float);
+      inputOffsets[i] = ringBuffers[i].read(inputOffsets[i], maxLag, samples.asBytes(), i > 0);
+    }
 
-      KJ_AVCALL(avcodec_send_frame(codecCtx, frame));
+    KJ_AVCALL(avcodec_send_frame(codecCtx, frame));
 
-      totalSamples += frame->nb_samples;
-      samplesSinceFlush += frame->nb_samples;
-      frame->pts = totalSamples * frame->time_base.den / frame->sample_rate;
-    }
+    totalSamples += frame->nb_samples;
+    samplesSinceFlush += frame->nb_samples;
 
-    // Note to self: To trigger EOF, do:
-    //   KJ_AVCALL(avcodec_send_frame(codecCtx, nullptr));
+    // This is a hack. Setting PTS correctly would look like this:
+    //
+    // frame->pts = totalSamples * TIMESTAMP_SCALE / frame->sample_rate;
+    //
+    // However DTS must be adjusted accordingly to avoid "Queue input is backward in time"
+    // warnings while streaming. Tweak both below, before sending each packet.
+    frame->pts = totalSamples;
```

Comment:
This is a little hacky, but I couldn't figure out a way to make `libopus` happy otherwise. This should just be:

```
frame->pts = totalSamples * TIMESTAMP_SCALE / frame->sample_rate;
```

Without any of the frame PTS/DTS adjustments below. However, without it, the server log is riddled with `Queue input is backward in time` whose root cause I don't fully understand, but tweaking DTS alongside PTS avoids it.